### PR TITLE
[snmp] Configure snmp docker hostname from config DB

### PR DESCRIPTION
**Pull request by vsenchyshyn**
Signed-off-by: Vitaliy Senchyshyn <vsenchyshyn@barefootnetworks.com>

**- What I did**
Configure snmp docker hostname with the value received from the config DB in order to fix SNMP CT which failed because the "sonic" sysName was always received.

**- How I did it**
Configure snmp docker hostname in the same way as this is done in hostname-config.sh

**- How to verify it**
Run snmp CT and verify it will pass

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vsenchyshyn** [comment]
@lguohan @stcheng Please review

**Comment by qiluo-msft** [comment]
Could you clarify what is 'snmp CT'?

File: dockers/docker-snmp-sv2/start.sh [hunk 1]
```diff
@@ -9,6 +9,15 @@ sonic-cfggen -d -y /etc/sonic/snmp.yml -t /usr/share/sonic/templates/snmpd.conf.
 mkdir -p /var/sonic
 echo "# Config files managed by sonic-config-engine" > /var/sonic/config_status
 
+CURRENT_HOSTNAME=`hostname`
+HOSTNAME=`sonic-cfggen -d -v DEVICE_METADATA[\'localhost\'][\'hostname\']`
+
+echo $HOSTNAME > /etc/hostname
+hostname -F /etc/hostname
```
Comment (qiluo-msft) [inline dockers/docker-snmp-sv2/start.sh]: why not use `hostname $HOSTNAME`?

Comment (vsenchyshyn) [reply dockers/docker-snmp-sv2/start.sh]: > why not use `hostname $HOSTNAME`?

In this case /etc/hostname will contain the old hostname.

Comment (vsenchyshyn) [reply dockers/docker-snmp-sv2/start.sh]: @qiluo-msft  Or you meant only this line: "hostname -F /etc/hostname"? If yes, this can be changed.

Comment (qiluo-msft) [reply dockers/docker-snmp-sv2/start.sh]: I see. Let's keep as is.

---
In reply to: [274999450](https://github.com/Azure/sonic-buildimage/pull/2773#discussion_r274999450) [](ancestors = 274999450)

File: dockers/docker-snmp-sv2/start.sh [hunk 2]
```diff
@@ -9,6 +9,15 @@ sonic-cfggen -d -y /etc/sonic/snmp.yml -t /usr/share/sonic/templates/snmpd.conf.
 mkdir -p /var/sonic
 echo "# Config files managed by sonic-config-engine" > /var/sonic/config_status
 
+CURRENT_HOSTNAME=`hostname`
+HOSTNAME=`sonic-cfggen -d -v DEVICE_METADATA[\'localhost\'][\'hostname\']`
```
Comment (qiluo-msft) [inline dockers/docker-snmp-sv2/start.sh]: >HOSTNAME [](start = 0, length = 8)

Will you handle command error or empty hostname cases?

Comment (vsenchyshyn) [reply dockers/docker-snmp-sv2/start.sh]: > > HOSTNAME [](start = 0, length = 8)
> 
> Will you handle command error or empty hostname cases?

Added validation. Please check.

**Comment by vsenchyshyn** [comment]
@qiluo-msft Could you please merge it?

File: dockers/docker-snmp-sv2/start.sh [hunk 3]
```diff
@@ -9,6 +9,17 @@ sonic-cfggen -d -y /etc/sonic/snmp.yml -t /usr/share/sonic/templates/snmpd.conf.
...
+
+if [ "$?" == "0" ] && [ "$HOSTNAME" != "" ]; then
+    echo $HOSTNAME > /etc/hostname
+    hostname -F /etc/hostname
+
+    sed -i "/\s$CURRENT_HOSTNAME$/d" /etc/hosts
```
Comment (qiluo-msft) [inline dockers/docker-snmp-sv2/start.sh]: >sed -i "/\s$CURRENT_HOSTNAME$/d" /etc/hosts [](start = 4, length = 43)

This line does not work inside docker container.
ref: [http://blog.jonathanargentiero.com/docker-sed-cannot-rename-etcsedl8ysxl-device-or-resource-busy/](http://blog.jonathanargentiero.com/docker-sed-cannot-rename-etcsedl8ysxl-device-or-resource-busy/)

Could you double check and fix?

**Pull request merged**